### PR TITLE
enh(resource): add new architecture to MetaService details

### PR DESCRIPTION
**Pull request by adr-mo**
Refacto of the detail metaService step one (architecture + tests)

### CODE REVIEW

File: src/Core/Application/Configuration/Repository/ReadMetaServiceRepositoryInterface.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ * For more information : contact@centreon.com
+ *
+ */
+declare(strict_types=1);
+
+namespace Core\Application\Configuration\Repository;
+
+use Core\Domain\Configuration\Model\MetaService;
+
+interface ReadMetaServiceRepositoryInterface
```
Comment (callapa) [inline src/Core/Application/Configuration/Repository/ReadMetaServiceRepositoryInterface.php:27]: I think you could create the subdirectory 'MetaService' in the 'Configuration' directory. (src/Core/Application/Configuration/**MetaService**/Repository/ReadMetaServiceRepositoryInterface.php

File: src/Core/Application/RealTime/UseCase/FindMetaService/FindMetaService.php [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+    ): void {
+        $this->info(
+            "Searching details for Meta Service",
+            [
+                "id" => $metaId
+            ]
+        );
+
+        if ($this->contact->isAdmin()) {
+            /**
```
Comment (callapa) [inline src/Core/Application/RealTime/UseCase/FindMetaService/FindMetaService.php:81]:
```suggestion
            $this->debug('Find MetaService as an admin user');
            /** 
```

File: src/Core/Application/RealTime/UseCase/FindMetaService/FindMetaService.php [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+                return;
+            }
+
+            $metaService = $this->repository->findMetaServiceById($metaId);
+            if ($metaService === null) {
+                $this->handleMetaServiceNotFound($metaId, $presenter);
+                return;
+            }
+        } else {
+            $accessGroups = $this->accessGroupRepository->findByContact($this->contact);
```
Comment (callapa) [inline src/Core/Application/RealTime/UseCase/FindMetaService/FindMetaService.php:96]:
```suggestion
            $this->debug('Find MetaService as an non-admin user');
            $accessGroups = $this->accessGroupRepository->findByContact($this->contact);
```

File: src/Core/Application/RealTime/UseCase/FindMetaService/FindMetaService.php [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+        FindMetaServicePresenterInterface $presenter
+    ): void {
+        $this->error(
+            "Meta Service configuration not found",
+            [
+                'id' => $metaId,
+                'userId' => $this->contact->getId()
+            ]
+        );
+        $presenter->setResponseStatus(new NotFoundResponse('MetaService'));
```
Comment (callapa) [inline src/Core/Application/RealTime/UseCase/FindMetaService/FindMetaService.php:151]:
```suggestion
        $presenter->setResponseStatus(new NotFoundResponse('MetaService configuration'));
```

File: src/Core/Domain/Configuration/Model/MetaService.php [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+    public function getRegexpSearchServices(): ?string
+    {
+        return $this->regexpSearchServices;
+    }
+
+    /**
+     * @param string|null $regexpSearchServices
+     * @return self
+     */
+    public function setregexpSearchServices(?string $regexpSearchServices): self
```
Comment (callapa) [inline src/Core/Domain/Configuration/Model/MetaService.php:159]:
```suggestion
    public function setRegexpSearchServices(?string $regexpSearchServices): self
```

File: src/Core/Domain/Configuration/Model/MetaService.php [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+    public function isActivated(): bool
+    {
+        return $this->isActivated;
+    }
+
+    /**
+     * @param bool $isActivated
+     * @return self
+     */
+    public function setIsActivated(bool $isActivated): self
```
Comment (callapa) [inline src/Core/Domain/Configuration/Model/MetaService.php:231]:
```suggestion
    public function setActivated(bool $isActivated): self
```

**Review by callapa — CHANGES_REQUESTED** [review]

File: src/Core/Application/Configuration/Repository/ReadMetaServiceRepositoryInterface.php (hunk 1 above)
Comment (adr-mo) [reply src/Core/Application/Configuration/Repository/ReadMetaServiceRepositoryInterface.php]: Done ✅

File: src/Core/Application/RealTime/UseCase/FindMetaService/FindMetaService.php (hunk 2 above)
Comment (adr-mo) [reply src/Core/Application/RealTime/UseCase/FindMetaService/FindMetaService.php:82]: Done ✅

File: src/Core/Application/RealTime/UseCase/FindMetaService/FindMetaService.php (hunk 3 above)
Comment (adr-mo) [reply src/Core/Application/RealTime/UseCase/FindMetaService/FindMetaService.php:98]: Done ✅

File: src/Core/Application/RealTime/UseCase/FindMetaService/FindMetaService.php (hunk 4 above)
Comment (adr-mo) [reply src/Core/Application/RealTime/UseCase/FindMetaService/FindMetaService.php]: Done ✅

File: src/Core/Domain/Configuration/Model/MetaService.php (hunk 5 above)
Comment (adr-mo) [reply src/Core/Domain/Configuration/Model/MetaService.php]: Done ✅

File: tests/php/Core/Domain/Configuration/MetaServiceTest.php [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+declare(strict_types=1);
+
+namespace Tests\Core\Domain\Configuration\Model;
```
Comment (kduret) [inline tests/php/Core/Domain/Configuration/MetaServiceTest.php:23]: namespace does not match file name

Comment (adr-mo) [reply tests/php/Core/Domain/Configuration/MetaServiceTest.php:23]: omg

File: src/Core/Infrastructure/Configuration/Repository/MetaService/DbReadMetaServiceRepository.php [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+declare(strict_types=1);
+
+namespace Core\Infrastructure\RealTime\Repository\MetaService;
```
Comment (callapa) [inline src/Core/Infrastructure/Configuration/Repository/MetaService/DbReadMetaServiceRepository.php:23]: Wrong namespace

**Review by callapa — CHANGES_REQUESTED** [review]

**Review by kduret — APPROVED** [review]

**Review by callapa — APPROVED** [review]

**Pull request merged**